### PR TITLE
Adds skeleton docs build

### PR DESCRIPTION
**Pull request by breakreturn**
A minimal buildable setup with license info, build
instructions and contribution guide.

Signed-off-by: Joakim Gross <joakim.gross@pelagicore.com>

### CODE REVIEW

File: scripts/publish_docs.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# SPDX-License_identifier: LGPL-2.1
+# This file is subject to the terms of the LGPL-2.1 license.
+# Please see the LICENSE file for details.
+
+set -e
+
+### Abort if the build is dirty
+if [[ "`git diff --shortstat 2> /dev/null | tail -n1`" != "" ]]
+then
+    echo "Aborting pulication of documentation due to the repository being 'dirty'"
```
Comment (frznlogic) [inline scripts/publish_docs.sh]: pulication

**Review by frznlogic — CHANGES_REQUESTED** [review]
Do we want to maintain a Vagrantfile as well for creating maintained build environments as well?

@jeremiah says; I would say yes! Definitely!

**Review by rpannek — APPROVED** [review]

**Review by DunderRoffe — APPROVED** [review]
Using CMake seems overkill but it looks fine

File: scripts/publish_docs.sh (hunk 1 above)
Comment (jeremiah) [reply scripts/publish_docs.sh]: I got pulicated!

**Review by jeremiah — APPROVED** [review]
* I'm missing a definition of what SWF is for a potential user who has never heard the name before.
* I added some boilerplate and a date to the README

**Comment by DunderRoffe** [comment]
As discussed offline, using pure Sphinx or python might be an alternative to cmake.
In the Sphinx documentation they refer to building using the following command in bash:

```bash
$ sphinx-build -b html sourcedir builddir
```
http://www.sphinx-doc.org/en/stable/tutorial.html

**Comment by breakreturn** [comment]
I personally feel that a VM is overkill for this build. There are no tests or other activities that would make a common environment beneficial, and the dependencies are quite reasonable for developers of this to have installed. I don't foresee any issues with the CI machines lacking the dependencies either. Perhaps we can have a VM as an improvement in the future if we see a need for it?

**Comment by breakreturn** [comment]
Yes, I just reused previous stuff and based it on CMake for no good reason I think. I agree pure Python would be easier. Can we treat that as a future improvement as well?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,41 @@
...
+Software Factory is maintained at https://github.com/Pelagicore/software-factory
+
+Maintainer: Joakim Gross <joakim.gross@pelagicore.com>
+
+
+## Dependencies
+* Sphinx
+* sphinxcontrib-seqdiag
+* sphinxcontrib-blockdiag
+* sphinxcontrib-actdiag
```
Comment (erikboto) [inline README.md:14]: I also needed to install sphinxcontrib-manpage

Comment (breakreturn) [reply README.md:14]: True. Will fix.

Comment (erikboto) [reply README.md:14]: Oh, and I needed texlive-latex-base (on Debian) for pdflatex which is used for generating PDFs

**Review by frznlogic — APPROVED** [review]

**Comment by frznlogic** [comment]
Your argument makes sense regarding build environment @breakreturn . If we see a need for it in the future ti can be added.

**Review by erikboto — APPROVED** [review]

**Pull request merged**